Release 3.2.0 alpha.1 (#5336)
open-formulieren/open-forms: CHANGELOG.rst

.. warning::

   The Camunda registration backend will be removed in Open Forms 4.0. There is no
   replacement scheduled - if you rely on this plugin, please get in touch.

.. warning::

    For the generic JSON registration plugin, we changed the way that the data is generated.
    In case of key conflicts between static, component, and user-defined variables,
    the static variables will take precedence. Previously, the component and user-defined
    variables would override the static variables. Our validation guards against the use of
    keys that are already present in the static variables, but this does not cover old forms
    and newly-added static variables.

.. warning:: Manual intervention required

    In the 3.1.1 bugfix release we fixed a bug regarding the default values of some components
    being ``null``. We added a script to fix any forms that still might be affected by these
    issues. You should run this script after deploying the patch release, to make sure the
    default values of affected components are fixed.

    .. code-block:: bash

        # in the container via ``docker exec`` or ``kubectl exec``:

>         python /app/bin/fix_component_default_values.py

Reviewer: is this still relevant here? It specifically mentions "the patch release" but this is an alpha release :)

**New features**

> * [:backend:`4933`] Added support for adding Cosign V2 information to registration email templates.

Reviewer: for consistency with the patch release notes, I'd put this in the bugfixes section :grimacing:

> * [:backend:`5140`] Extend authentication module capability, which in the future will enable support for eIDAS and Yivi.

Reviewer: ```suggestion
* [:backend:`5140`] Reworked the authentication module architecture to make it possible to add support for
  new plugins based on the OpenID Connect protocol (Yivi and eIDAS).
```

> * [:backend:`5283`] UX: cleaned up columns of the form list view.

Reviewer: ```suggestion
* [:backend:`5283`] Cleaned up the displayed columns in the admin form list to improve the UX.
```

* [:backend:`5254`] Added new family-members prefill plugin.

    - The data can be retrieved from "Haal Centraal BRP personen bevragen" (version 2) or "StUF-BG" (version 3.1).
    - Partners or children of the authenticated user can be stored in a user-defined variable.
    - The retrieved data of children can be filtered by age and whether they are deceased.

> * [:backend:`4923`] Added JSON schema generation of a form through an API endpoint.

Reviewer: ```suggestion
* [:backend:`4923`] Added support for JSON schema generation of a form in the API.
```

    - The schema represents the submission data and includes all user-defined and component variables.

>     - The component schemas include validation rules and a description if available.

Reviewer: ```suggestion
    - The component schemas include validation rules and a description when available.
```

**Bugfixes**

> * [:backend:`5300`] Fixed nested submission data not being saved.

Reviewer: ```suggestion
* [:backend:`5300`] Fixed a regression in the previous alpha release where nested submission data was not being saved.
```
added a bit more context to avoid panic for people currently on 3.1 :sweat_smile:

**Project maintenance**

> * [:backend:`5285`] Switched to django-structlog from logging.

Reviewer: ```suggestion
* [:backend:`5285`] Improved the logs emitted by the application to better integrate with observability tooling like Grafana.
```
this one can definitely be moved into the features!
